Detectors/TPC/workflow/src/FileReaderWorkflow.cxx

```diff
@@ -0,0 +1,107 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   FileReaderWorkflow.cxx
+
+#include "TPCWorkflow/PublisherSpec.h"
+#include "TPCWorkflow/TrackReaderSpec.h"
+
+#include "Algorithm/RangeTokenizer.h"
+
+#include "SimulationDataFormat/IOMCTruthContainerView.h"
+#include "SimulationDataFormat/ConstMCTruthContainer.h"
+#include "SimulationDataFormat/MCCompLabel.h"
+
+// add workflow options, note that customization needs to be declared before
+// including Framework/runDataProcessing
+void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
+{
+  using namespace o2::framework;
+
+  std::vector<ConfigParamSpec> options{
+    {"input-type", VariantType::String, "clusters", {"clusters, tracks"}},
+    {"disable-mc", VariantType::Bool, false, {"disable sending of MC information"}}};
+
+  std::swap(workflowOptions, options);
+}
+
+#include "Framework/runDataProcessing.h" // the main driver
+
+using namespace o2::framework;
+
+enum struct Input { Clusters,
+                    Tracks
+};
+
+const std::unordered_map<std::string, Input> InputMap{
+  {"clusters", Input::Clusters},
+  {"tracks", Input::Tracks}};
```

Comment:
This does not necessarily need to be there. I could also include the RecoWorkflow header and use the `OutputType` enum from there instead and adjust the `InputMap` but I think this is the nicer solution.

Reply:
I think this is ok, it is not so much code... :)
When you are ready, can you remove the WIP such that the CI runs, or is there still something to update?